src/outpack/util.py

```diff
@@ -200,3 +200,30 @@ def openable_temporary_file(*, mode: str = "w+b", dir: Optional[str] = None):
             os.unlink(f.name)
         except OSError:
             pass
+
+
+@overload
```

Comment:
are these bits just for mypy?

Reply:
Yeah, it makes it possible to describe that the output type depends on the input. Might be overkill in this context